astromodels/functions/functions_2D.py

```diff
@@ -95,7 +95,23 @@ def get_boundaries(self):
         max_lon = max(_coord.transform_to("icrs").ra.value)
 
         return (min_lon, max_lon), (min_lat, max_lat)
+        
+    def get_total_spatial_integral(self, z=None):  
+        """
+        Returns the total integral (for 2D functions) or the integral over the spatial components (for 3D functions).
+        needs to be implemented in subclasses.
+
+        :return: an array of values of the integral (same dimension as z).
+        """
+
+        dL= l_max-l_min if l_max > l_min else 360 + l_max - l_max
```

Comment:
Looks like there are a bunch of undefined variables in this method (`l_max`, `l_min`, `sigma_b`...). Did you test it?